lib/board.rb

```diff
@@ -68,10 +68,24 @@ def valid_placement?(ship, coordinates)
     end
   end
 
+  def overlap_check(ship, coordinates)
+    ships = coordinates.map do |coordinate|
+      cells[coordinate].ship
+    end
+    if ships.all? {|ship|  ship == nil}
+      true
+    else
+      false
+    end
+  end
+  
```

Comment:
Good method to use within valid_placement method

Reply:
Agreed this will resolve the error, and pass the test.

Reply:
Okay Ill approve this review